### DISCUSSION TITLE
Q: Am I totally of the grid now? (HTTP connection)

### DISCUSSION THREAD

**Discussion by Simension** [krahabb/meross_lan#418] [Q&A]
Hi guys,
I installed a MSS210 smartplug with the help of the official Meross app, so the device is immediately detected in HA afterwards. Then in HA I used my account-credentials to fetch my devicekey and to connect the smartplug to HA.
I deleted the same Smartplug from HA, reinstalled it not using my credentials but only the device-key.
I even deleted the smartplug out of the official Meross app and the internet-connection is blocked for the plug in router-settings. In HA I configured it to only http. Its working perfectly fine and is only appearing in HA.
Is the Meross-Server now used in any way or am I totally off the grid and using it only local? In the docs its said "The integration will just 'side-communicate' over HTTP". So "side-communicate" means only local?
Sorry if I get anything wrong. Iam new to HA and to smarthomestuff. I just want to use it without any external servers which could track me.
Thank you guys!

P.S: I found this thread which seems to answer my question (https://github.com/krahabb/meross_lan/discussions/322). But perhaps the slightly different way I did it and the following discussion could be also a help for others who want a easy way to go local. So I will start this thread anyway if this is okay for you guys (please tell me if this is not okay, i will delete then).

**Comment by krahabb** [comment]
Hello @Simension,
As for the device, having blocked any internet-bound traffic it is only locally available  (if we trust the router;).
As for `meross_lan`, you just have to disable/delete any 'Meross cloud profile' entry so that meross_lan will not attempt any outbound traffic which would anyway be useless since the device, being firewalled, doesn't have cloud-mqtt connectivity.

Fixing protocol to HTTP for the device configuration is the last step, just a refinement, but this step, by itself, would not prevent meross_lan to connect to the cloud mqtt broker since this connection is established when you setup the aforementioned 'Meross cloud profile' regardless of any device configuration

**Comment by Simension** [comment]
Thanks for further information krahabb! (And thanks for integration, works perfectly fine! Its a pleasure!) Oright, deleted the Meross cloud profile already which was also visible under the hubs segment. 
By the way for all guys who have a local dns-resolver like pihole: Even if internet-connection is blocked for the smartplug it will send dns-queries like "mqtt-eu-4.meross.com" to pihole because it is local. They will be answered.
